src/common/meta/src/heartbeat/utils.rs

```diff
@@ -56,3 +57,62 @@ pub fn outgoing_message_to_mailbox_message(
         )),
     })
 }
+
+/// Extracts datanode workloads from the provided optional `NodeWorkloads`.
+///
+/// Returns default datanode workloads if the input is `None`.
+pub fn get_datanode_workloads(node_workloads: Option<&NodeWorkloads>) -> DatanodeWorkloads {
+    #[cfg(feature = "enterprise")]
+    match node_workloads {
+        Some(NodeWorkloads::Datanode(datanode_workloads)) => datanode_workloads.clone(),
+        _ => DatanodeWorkloads {
+            types: vec![DatanodeWorkloadType::Hybrid as i32],
+        },
+    }
+
+    #[cfg(not(feature = "enterprise"))]
+    match node_workloads {
```

Comment:
Add a log here if non hybrid workload given but we only have hybrid in this version

Reply:
Resolved in [54f877a](https://github.com/GreptimeTeam/greptimedb/pull/6055/commits/54f877a46d599738d570eaf7c984be163f115105)